File: find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int>v,ans;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==target){
                v.push_back(i);
            }
        }
        if(v.size()==1){
            ans.push_back(v[0]);
            ans.push_back(v[0]);
        }else if(v.empty()){
            ans.push_back(-1);
            ans.push_back(-1);
        }else{
            ans.push_back(v[0]);
            ans.push_back(v[v.size()-1]);
        }
        return ans;
    }
};

File: find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int i =0, j = nums.length -1;
        while(i <= j){
            int mid = i + (j -i)/2;
            if(nums[mid] < target){
                i = mid + 1;
            }else if(nums[mid] > target){
                j = mid - 1;
            }else{
                int lo = mid;
                int hi = mid;
                while(lo > 0 && nums[lo - 1] == target){
                    lo--;
                }
                while(hi < nums.length - 1 && nums[hi + 1] == target){
                    hi++;
                }
                return new int[]{lo, hi};
            }
        }
        return new int[]{-1,-1};
    }
}